db/migrate/20240617132532_cleanup_system_roles_and_access_controls.rb

```diff
@@ -0,0 +1,59 @@
+class CleanupSystemRolesAndAccessControls < ActiveRecord::Migration[7.0]
+  # Cleanup a bug in the system scope on Role
+  def up
+    [
+      'System Role - Can Edit Project Groups',
+      'System Role - Can Participate in Cohorts',
+      'System Role - Can View Cohorts',
+    ].each do |name|
+      keep_id = Role.where(name: name, system: true).minimum(:id)
+      delete_ids = Role.where(name: name, system: true).where.not(id: keep_id).pluck(:id)
+      # Find the unique access controls based on collection_id, and user_group_id that use the
+      # roles in deleted_ids.
+      # Keep the first of each of those, updating the role_id to be keep_id, unless one already exists
+      # using role_id: keep_id
+      # Remove any Access Controls that use delete_ids
+      combinations = {}
+      AccessControl.where(role_id: delete_ids).find_each do |ac|
+        combinations[[ac.collection_id, ac.user_group_id]] ||= []
+        combinations[[ac.collection_id, ac.user_group_id]] << ac
+      end
+      combinations.each do |(collection_id, user_group_id), access_controls|
+        # We already have one that uses the role we want to keep
+        next if access_controls.any? { |ac| ac.role_id == keep_id }
+
+        puts "Found #{access_controls.count} Access Controls that should have been using the role #{keep_id}"
+        access_controls.min_by(&:id).update(role_id: keep_id)
+      end
+      AccessControl.where(role_id: delete_ids).destroy_all
+      Role.where(id: delete_ids).destroy_all
+    end
+
+    # Cleanup any exact duplicate Access Controls (this is probably just a side-effect of running the
+    # migration a bunch of times, but there's no harm making sure)
+    controls = AccessControl.system.group_by { |ac| [ac.collection_id, ac.user_group_id, ac.role_id] }
```

Comment:
Maybe there should be a unique constraint to prevent duplicates like this?

Reply:
I'll add that to the backlog.  Thanks.